pkg/websocket/server.go

```diff
@@ -107,28 +107,62 @@ func listenToWSServer(ctx context.Context, conn *gwebsocket.Conn) error {
 			continue
 		}
 
+		// ensure the environment is set up correctly
+		os.Setenv("KUBECONFIG", runtimeconfig.PathToKubeConfig())
+		os.Setenv("TMPDIR", runtimeconfig.EmbeddedClusterTmpSubDir())
+
 		switch msg.Command {
+		case "upgrade-manager":
```

Comment:
could you move this command into its own function? and perhaps in a follow up move the others?